connection.go

```diff
@@ -0,0 +1,131 @@
+package papergres
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// SSLMode defines all possible SSL options
+type SSLMode string
+
+const (
+	// SSLDisable - No SSL
+	SSLDisable SSLMode = "disable"
+	// SSLRequire - Always SSL, no verification
+	SSLRequire SSLMode = "require"
+	// SSLVerifyCA - Always SSL, verifies that certificate was signed by trusted CA
+	SSLVerifyCA SSLMode = "verify-ca"
+	// SSLVerifyFull - Always SSL, verifies that certificate was signed by trusted CA
+	// and server host name matches the one in the certificate
+	SSLVerifyFull SSLMode = "verify-full"
+)
+
+// Connection holds all database connection configuration.
+type Connection struct {
+	Database    string
+	User        string
+	Password    string
+	Host        string
+	Port        string
+	AppName     string
+	Timeout     int
+	SSLMode     SSLMode
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// NewConnection creates and returns the Connection object to the postgres server
+func NewConnection(databaseURL string, appName string) Connection {
+	u, err := url.Parse(databaseURL)
+	if err != nil {
+		panic(err)
+	}
+
+	host, port, _ := net.SplitHostPort(u.Host)
+	p, _ := u.User.Password()
+	q, _ := url.ParseQuery(u.RawQuery)
```

Comment:
In this case `q, _ := url.ParseQuery(u.RawQuery)` `_` is an actual error but it's ok to not read into it since if the connection string or URI  passed is incorrect or incompatible or there is a decoding error then, it will be caught on [Line#43]
 by the `Parse` method and we'll return an error before even reaching here on [Line#50].

```go
u, err := url.Parse(databaseURL)
if err != nil {
    panic(err)
}
```

The `ParseQuery` always returns a non-nil map containing all the valid query parameters found; error describes the first decoding error encountered, if any. So, the `ParseQuery` function parses the URL-encoded query string and returns a map listing the values specified for each key.
 
Consider the following example:
 ```
var connectionURI = "postgres://alpha:secret@127.0.0.1:5432/testDB"
// or 
var connectionInfoString = "user=alpha password=secret host=127.0.0.1 port=5432 dbname=testDB"
```
If we run `ParseQuery` on either of the two lines, it will return:
```
host:127.0.0.1
port: 5432
database: testDB
user: alpha
password: secret
```
So, it's completely safe to not catch the error here because it'll be caught much above by the `Parse` function

Similarly on [Line#48], it's safe to not catch the error returned by `net.SplitHostPort` which returns an error if it's not able to catch host and port from the connection string. And then again we are safe guarded by the `Parse` method called above on [Line#43], which will error out these things regarding malformed connection strings. 

All in all, we hadn't called the Parse function before calling these two methods or if we are calling these functions independently then we would certainly have to catch the error. But since we are one step ahead and call Parse earlier, **it's safe to ignore these errors** for they'll never propagate and we save on these extra lines of code :)

[Line#43]: https://github.com/Paperchain/papergres/blob/overhaul/connection.go#L43
[Line#48]: https://github.com/Paperchain/papergres/blob/overhaul/connection.go#L48
[Line#50]: https://github.com/Paperchain/papergres/blob/overhaul/connection.go#L50